content/blog/dotnet-preview/index.md

```diff
@@ -0,0 +1,233 @@
+
+## How Pulumi .NET Works
+
+Pulumi is declarative even though it uses general purpose programming languages. It does this by declaring a set of resources, and the Pulumi engine orchestrates the CRUD operations to build and deploy infrastructure.
+
+A Pulumi project uses a .NET Core console application to declare infrastructure. To illustrate how this is accomplished, let's create an Azure App Service with an SQL database. The first step is to make sure that your environment meets the prerequisites:
+
+### Prerequisites
+
+1. [Install Pulumi](https://www.pulumi.com/docs/get-started/install/)
+
+1. [Install .NET Core 3.0+](https://dotnet.microsoft.com/download)
+
+1. [A Microsoft Azure account](https://azure.microsoft.com/en-us/free/)
+
+1. [Install Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli)
+
+### Get the Examples
+
+Pulumi provides a repository of examples to get you started. Clone the examples from GitHub.
+
+   ``` bash
+   $ git clone https://github.com/pulumi/examples/
+   $ cd examples/
+   ```
+
+### Create a New Stack
+
+Change the directory to `azure-cs-appservice` directory
+
+   ``` bash
+   $ cd azure-cs-appservice
+   ```
+
+There are four files in the Pulumi project, the `Azure.Appservice.csproj` which sets the .NET SDK and the Pulumi packages for the project. `Pulumi.yaml` holds project parameters such as name and runtime. The `Pulumi.dev.yaml` file holds the database password. `SharedAccessSignature.cs` contains a helper function to generate storage blob access URLs with SAS tokens in them. It demonstrates that you can add custom code inside your Pulumi program. `Program.cs` is the main program for declaring resources.
+
+1. Create a new stack:
+
+    ``` bash
+    $ pulumi stack init dev
+    ```
+
+1. Login to Azure CLI (you will be prompted to do this during deployment if you forget this step):
+
+    ``` bash
+    $ az login
+    ```
+
+1. Configure the location to deploy the resources to:
+
+    ``` bash
+    $ pulumi config set azure:location centralus
+    ```
+
+1. Define SQL Server password (make it complex enough to satisfy Azure policy):
+
+    ``` bash
+    pulumi config set --secret sqlPassword <value>
+    ```
+
+1. Run `pulumi up` to preview and deploy changes:
+
+    ``` bash
+    $ pulumi up
+    Previewing changes:
+    ...
+
+    Performing changes:
+    ...
+    info: 10 changes performed:
+
+        + 10 resources created
+
+    Update duration: 1m14.59910109s
+    ```
+
+___
+**NOTE:**
+
+> If you are using a free Azure account, some regions do not allow creating services if they are too busy. You might have to try a different [Azure location](https://azure.microsoft.com/en-us/global-infrastructure/locations/). You must tear down the stack resources to update the project's location information.
+___
+
+### Test Your Stack
+
+Check the deployed website endpoint by querying the stack and using curl to retrieve the web page.
```

Comment:
Perhaps add a note that `endpoint` is available because it was one of the stack outputs returned from the program.

Reply:
added